bin/report

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+# bin/report <build-dir> <cache-dir> <env-dir>
+
+### Configure environment
+
+set -o errexit    # always exit on error
+set -o pipefail   # don't ignore exit codes when piping output
+
+BUILD_DIR=${1:-}
+
+# <separator> <array>
+joinArrayWithSeparator() {
+  local IFS="$1"
+  shift
+  echo "$*"
+}
+
+packages=()
+custom_packages=()
+custom_repositories=()
+
+while IFS= read -r line; do
+  if [[ $line ]]; then
+    if echo "${line}" | grep -q -e "^:repo:"; then
+      custom_repositories+=("${line//:repo:/}")
+    elif [[ $line == *deb ]]; then
+      custom_packages+=("${line}")
+    else
+      packages+=("${line}")
+    fi
+  fi
+done < <(grep -v -s -e '^#' < "${BUILD_DIR}/Aptfile")
```

Comment:
Do you really want `-s` here?

Reply:
now that you point it out, i don't think so. strangely, i got the same output when i tested this script with and without the `-s` flag against a target Aptfile even though running the singular `grep` command produces different results.

Reply:
i'm going to replace this line with `done < "${BUILD_DIR}/Aptfile"` to process all the lines instead of trying to filter out comment lines eagerly.